File: day7/Day7.java
class Day7 {
    public int countElements(int[] arr) {
        
        int n = arr.length;
        
        Map<Integer,Integer>map = new HashMap<Integer,Integer>();
        
        for(int i = 0; i < n;i++) {
            
            if(map.containsKey(arr[i]) == false) {
                map.put(arr[i],1);
            }
            else {
                map.put(arr[i],map.get(arr[i]) + 1);
            }
        }
        int ans = 0;
        for(Map.Entry<Integer,Integer> entry : map.entrySet()) {
            
            Integer key = entry.getKey();
            Integer count = entry.getValue();
            
            if(map.containsKey(key+1) == true) {
                ans += (int)count;
            }
        }
        return ans;
    }
}

File: day7/day7.cpp
class day7 {
public:
    int countElements(vector<int>& arr) {
        
        int n = arr.size();
        unordered_map<int,int>hash;
        
        for(int i = 0 ; i < n;i++) {
            if(hash.find(arr[i]) == hash.end()) {
                hash[arr[i]] = 1;
            }
            else {
                hash[arr[i]]++;
            }
        }
        int ans = 0;
        for(auto it = hash.begin();it != hash.end();it++) {
            
            int key = it->first;
            int count = it->second;
            if(hash.find(key+1) != hash.end()) {
                ans += count;
            }
        }
        return ans;
    }
};